tweet_recommendations/recommendation.py

```diff
@@ -11,32 +11,46 @@
 
 
 def embedding_similarity(x: np.ndarray, y: np.ndarray):
-    similarity = cosine_similarity(x, y)
+    similarity = np.clip(cosine_similarity(x, y), -1, 1)
     ang_dist = np.arccos(similarity) / np.pi
     ang_sim = 1 - ang_dist
     return ang_sim
 
 
 def normalise(array: np.ndarray):
-    return (array - array.min()) / (array.max() - array.min())
+    if array.max() != array.min():
```

Comment:
Jeśli są to floaty, to warunek będzie prawdziwy co najwyżej z przypadku (na wbudowanym typie floata chyba nie ma takich przypadków). Z drugiej strony różnica może dać 0 przez co poniższe się wysypie. To jest odczucie, nie musi się wcale tak zdarzyć, ale proponowałbym użycie `np.isclose(array.max(), array.min(), rtol=1e-8)`.